tests/bsim/bluetooth/host/security/bond_per_connection/src/bs_bt_utils.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+
+BUILD_ASSERT(CONFIG_BT_MAX_PAIRED >= 2, "CONFIG_BT_MAX_PAIRED is too small.");
+BUILD_ASSERT(CONFIG_BT_ID_MAX >= 3, "CONFIG_BT_ID_MAX is too small.");
+
+#define BS_SECONDS(dur_sec)    ((bs_time_t)dur_sec * 1000000)
+#define TEST_TIMEOUT_SIMULATED BS_SECONDS(60)
+
+void test_tick(bs_time_t HW_device_time)
+{
+	bs_trace_debug_time(0, "Simulation ends now.\n");
+	if (bst_result != Passed) {
+		bst_result = Failed;
+		bs_trace_error("Test did not pass before simulation ended.\n");
+	}
+}
+
+void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(TEST_TIMEOUT_SIMULATED);
+	bst_result = In_progress;
+}
+
+DEFINE_FLAG(flag_is_connected);
+struct bt_conn *g_conn;
+DEFINE_FLAG(bondable);
+
+void wait_connected(void)
+{
+	WAIT_FOR_FLAG(flag_is_connected);
+}
+
+void wait_disconnected(void)
+{
+	WAIT_FOR_FLAG_UNSET(flag_is_connected);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	UNSET_FLAG(flag_is_connected);
+}
+
+BUILD_ASSERT(CONFIG_BT_MAX_CONN == 1, "This test assumes a single link.");
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	ASSERT((!g_conn || (conn == g_conn)), "Unexpected new connection.");
+
+	if (!g_conn) {
+		g_conn = bt_conn_ref(conn);
+	}
+
+	if (err != 0) {
+		clear_g_conn();
+		return;
+	}
+
+	SET_FLAG(flag_is_connected);
+
+	if (bt_conn_set_bondable(conn, bondable)) {
```

Comment:
could you add another round of bonding without calling `bt_conn_set_bondable`?
like have a bondable by default device, try bonding and see that it works and then continue with the two test rounds you already have

Reply:
Added such case.